diff --git a/submodule-options.nix b/submodule-options.nix
index e48c041..a8e96fd 100644
--- a/submodule-options.nix
+++ b/submodule-options.nix
@@ -198,18 +198,7 @@ let
   ];
   dir = submodule ([
     commonOpts
     dirOpts
-    {
-      imports = [
-        (mkRemovedOptionModule
-          [ "method" ]
-          ''
-            ▹ persistence."${name}":
-                As real bind mounts are now used instead of bindfs, changing the directory linking
-                method is deprecated.
-          '')
-      ];
-    }
     (mkIf (homeDir != null) { home = homeDir; })
     ({ config, ... }:
       let
